File: Backend/app_administracion_medicamentos/src/main/java/app_administracion_medicamentos_backend/api/services/ClienteService.java
package app_administracion_medicamentos_backend.api.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import app_administracion_medicamentos_backend.api.dao.ClienteDao;
import app_administracion_medicamentos_backend.api.dto.entities.Cliente;
import app_administracion_medicamentos_backend.api.interfaces.services.IClienteService;

@Service
public class ClienteService implements IClienteService<Cliente>{
	
	@Autowired
	ClienteDao clienteDao;

	@Override
	public List<Cliente> getAll() {
		List<Cliente> clientes = this.clienteDao.findAll();
		return clientes;
	}

	@Override
	public Cliente getById(Long id) {
		return this.clienteDao.getById(id);
	}

	@Override
	public Cliente createOrUpdate(Cliente entity) {
		try {
			Cliente cliente = this.clienteDao.save(entity);
			return cliente;
		} catch (Exception e) {
			return null;
		}
	}

	@Override
	public Cliente deleteById(Long id) {
		Cliente cliente = getById(id);
		if(cliente != null) {
			this.clienteDao.deleteById(id);
			return cliente;
		}
		return null;
	}

}

File: Frontend/material-dashboard-angular2/src/app/programming/services/cliente.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Cliente } from '../dto/Cliente.dto';
import { IClienteService } from '../interfaces/IClienteService';
import { Response } from '../models/Response.dto';
import { environment } from 'environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ClienteService implements IClienteService<Cliente>{

  baseUrl: string = environment.baseUrl;
  getAllUrl: string = `${this.baseUrl}/api/proyecto/clientes/getAll`;
  getByIdUrl: string = `${this.baseUrl}/api/proyecto/clientes/getById`; // aqui falta el id
  createOrUpdateUrl: string = `${this.baseUrl}/api/proyecto/clientes/createOrUpdate`;
  deleteByIdUrl: string = `${this.baseUrl}/api/proyecto/clientes/deleteById`; // aqui falta el id

  constructor(private httpClient: HttpClient) { }

  getAll(): Observable<Response<Cliente[]>> {
    return this.httpClient.get<Response<Cliente[]>>(`${this.getAllUrl}`);
  }
  getById(id: number): Observable<Response<Cliente>> {
    return this.httpClient.get<Response<Cliente>>(`${this.getByIdUrl}/${id}`);
  }
  createOrUpdate(entity: Cliente): Observable<Response<Cliente>> {
    return this.httpClient.post<Response<Cliente>>(`${this.createOrUpdateUrl}`, entity);
  }
  deleteById(id: number): Observable<Response<Cliente>> {
    return this.httpClient.delete<Response<Cliente>>(`${this.deleteByIdUrl}/${id}`);
  }


}
